src/class/object_genome_variants.php

```diff
@@ -268,6 +254,24 @@ function checkFields ($aData, $zData = false, $aOptions = array())
             }
         }
 
+        foreach ($aData as $sField => $sVariant) {
+            if (preg_match('/DNA/', $sField)) {
+                // We want to check the input of all DNA fields: are these variant
+                //  descriptions indeed cleanly formatted? And if our check seems
+                //  to fail, is this perhaps because it holds syntax that we do
+                //  not support? If we DO support the syntax but the variant does
+                //  not seem to be HGVS, we will send an error.
+                $aVariantInfo = lovd_getVariantInfo($sVariant, false);
+                if (!lovd_isHGVS($sVariant)
+                    && !isset($aVariantInfo['errors']['ENOTSUPPORTED']) // Supported by LOVD.
+                    && !(isset($aVariant['warnings']['WNOTSUPPORTED'])  // Supported by VariantValidator.
+                        || isset($aVariant['messages']['IUNCERTAINPOSITIONS'])
+                        || isset($aVariant['messages']['IPOSTIONRANGE']))) {
+                    lovd_errorAdd($sField, 'The variant ' . $sVariant . ' did not pass our checks. Please take another look and try again.');
+                }
+            }
+        }
+
```

Comment:
Check the HGVS of all DNA input variants.

Reply:
I'm not sure if this works the way intended. If the variant is not HGVS but does have an `IUNCERTAINPOSITIONS`, then it's accepted? Not sure if that's OK.
Also, I think we can make this a bit more efficient by first doing the `lovd_isHGVS()` check, and running `lovd_getVariantInfo()` only when `lovd_isHGVS()` returns false?

Reply:
You were right!! It has been fixed.